src/api.rs

```diff
@@ -1230,6 +1244,23 @@ impl<T: Pixel> ContextInner<T> {
 
           update_rec_buffer(fi, fs);
 
+          // Copy the `rec_buffer` into subsequent frame invariants.
+          let rec_buffer = fi.rec_buffer.clone();
+          for subsequent_fi in self
+            .frame_invariants
+            .iter_mut()
+            .skip_while(|(&output_frameno, _)| {
+              output_frameno <= cur_output_frameno
+            })
+            .filter(|(_, fi)| !fi.skip)
+            .map(|(_, fi)| fi)
+          {
+            subsequent_fi.rec_buffer = rec_buffer.clone();
+            subsequent_fi.set_ref_frame_sign_bias();
```

Comment:
It would be really nice to track that these are invalid somehow, to avoid future bugs. Maybe tests are enough, I dunno. Or maybe they shouldn't even be a part of fi?

Reply:
The encode-decode tests caught the `rec_buffer` part but not the `ref_frame_sign_bias` part (which was caught by everything on AWCY). It'd be nice to remove them from FI (and potentially look into refactoring everything in FI that depends on cloning the previous frame's FI), but I can't really say how good of an idea it is because I'm not very familiar with the rest of the code yet.

Reply:
I thought maybe to make them Option, but they are used quite a bit. I guess we can leave it for the future.